File: client/src/app/models/order.ts
import { Address } from "./address";
import { Book } from "./book";
import { FilterParams } from "./pagination";
import { User } from "./user";

export interface OrderItem {
    id: number;
    quantity: number;
    book: Book;
}

export interface Order {
    id: number;
    code: string;
    amount: number;
    paymentStatus: string;
    orderStatus: string;
    orderDate: string;
    user: User;
}

export interface OrderDetail extends Order {
    subtotal: number;
    deliveryFee: number;
    paymentIntentId: string;
    shippingAddress: Address;
    items: OrderItem[];
}

export interface OrderParams extends FilterParams {
    paymentStatuses: string[];
    orderStatuses: string[];
    codeSearch?: string;
    userSearch?: string;
    minAmount?: number;
    maxAmount?: number;
    orderDateStart?: string;
    orderDateEnd?: string;
}

File: BookStoreNetReact.Domain/Entities/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations;

namespace BookStoreNetReact.Domain.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        [StringLength(32, MinimumLength = 32)]
        public string Code { get; set; } = Guid.NewGuid().ToString("N");
        [Range(0, double.MaxValue)]
        public double Subtotal { get; set; }
        [Range(0, int.MaxValue)]
        public int DeliveryFee { get; set; }
        [Range(0, double.MaxValue)]
        public double Amount { get; set; }
        public string Note { get; set; } = "";
        public required string PaymentIntentId { get; set; }
        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;
        public OrderStatus OrderStatus { get; set; } = OrderStatus.PendingConfirmation;
        public required ShippingAddress ShippingAddress { get; set; }
        [Range(1, int.MaxValue)]
        public required int UserId { get; set; }
        public AppUser? User { get; set; }
        public List<OrderItem> Items { get; set; } = new List<OrderItem>();
    }
}
